### ISSUE TITLE
DT-1177: Remove VM management from BLT

### ISSUE THREAD

**Issue by danepowell** [acquia/blt#3825]
This isn't set in stone, but I want to float the idea for BLT 11.x of removing all VM functionality from BLT. The primary reason is that Acquia Developer Studio now has a superior local development environment experience that includes managing DrupalVM as well as Lando, and even some nice features that BLT doesn't have such as one-step commands for adding services like Solr and Memcache.

`blt vm` has always been a bit of an oddball feature for BLT, because while the rest of BLT is designed to run in the same environment as a Drupal site (`blt setup`, `blt tests`, `blt sync`, etc... should all be run _inside_ the VM), `blt vm` is the only command that runs on the _host_ machine and manages the environment itself.

I think it would create a clearer separation of concerns and mental model for users if we removed `blt vm`, so that it's clear that BLT always run adjacent to a Drupal installation, and other tools like ADS should be used to manage the development environment itself.

**Comment by mikemadison13** [comment]
I'm not fundamentally opposed to this at all, especially given the fact that plugins can largely own the _recipes_ for these tasks (e.g. the current DrupalVM one or the ones that @lcatlett has been working on for Docksal and DDev). 

I do think the biggest concern I have with BLT becoming "less" opinionated on VMs is that DrupalVM has a different approach than most of the Docker solutions out there (one monolithic VM with many services, vs. separate containers for different services). 

TLDR it is good that BLT is moving towards a more VM agnostic approach, but necessarily that will require some of BLT to evolve to a more modern containerization support than it currently has. Behat tests are a great example, where BLT assumes that google-chrome is on the same VM as the web server.

**Comment by danepowell** [comment]
Yep the Behat / Chrome pain point is definitely something I'm aware of, and it's good to call out as an example of BLT will need more flexibility if it doesn't prescribe a VDE. Right now ADS works around it by maintaining a custom Lando image with Chrome baked-in, rather than running in a separate container as it really "ought" to.

**Comment by danepowell** [comment]
This would now be targeting BLT 12 / Drupal 9.
